### PR TITLE
Surf s3

### PR DESCRIPTION
**Pull request by Harry-Johnston**

### CODE REVIEW

File: public/js/populateDropdowns.js [hunk 1]
```diff
@@ -204,3 +206,24 @@ let outputBackgroundInfo = (options) => {
...
 };
+
+let checkPastAssessmentCheckbox = (futureDates, allDates) => {
+  pastAssessmentCheckbox.addEventListener('change', e => {
+    if (e.target.checked === true) {
+      outputAssessmentDates(allDates)
+    }
+    if (e.target.checked === false) {
+      outputAssessmentDates(futureDates)
+    }
```
Comment (ashleycoles) [inline public/js/populateDropdowns.js:217]: Make this into an else

File: src/Models/EventModel.php [hunk 2]
```diff
@@ -169,7 +169,7 @@ public function searchPastEvents(?string $searchTerm): array
      * @return array An array of Events based on category ID and previous X months from current date.
      */
 
-    public function getEventsByCategoryId(string $categoryId, int $previousMonths = 0): array
+    public function getEventsByCategoryId(string $categoryId, int $previousMonths = 9999): array
```
Comment (ashleycoles) [inline src/Models/EventModel.php:172]: I don't like this. Is there a better way?

**Review by ashleycoles — CHANGES_REQUESTED** [review]
Looking pretty good.

Seems to me like you've missed this requirement: 

If an applicant is assigned to a past date when editing them. The date should display as a uneditable text above the drop down

File: src/Models/EventModel.php (hunk 2 above)
Comment (ashleycoles) [reply src/Models/EventModel.php:172]: Would suggest leaving the original default, but pass in the number of previous months you want when this method gets called in the controller.

Maybe don't go for 9999, maybe do like 5 years worth?

File: templates/applicantForm.phtml [hunk 3]
```diff
@@ -133,9 +133,12 @@
         </div>
 
         <div id="assessment" class="collapse">
-            <label for="assessmentDay">Assessment day</label>
+            <label for="assessmentDay">Assessment day
+                <label for="pastAssessmentDays">
+                    <input type="checkbox" id="pastAssessmentDays" name="accept" value="yes">  Include past assessment days
```
[the remark this answers is not in the document]
Comment (Harry-Johnston) [reply templates/applicantForm.phtml:138]: done

File: public/js/populateDropdowns.js (hunk 1 above)
Comment (Harry-Johnston) [reply public/js/populateDropdowns.js:217]: done

File: src/Models/EventModel.php (hunk 2 above)
Comment (Harry-Johnston) [reply src/Models/EventModel.php:172]: done

File: templates/applicantForm.phtml [hunk 4]
```diff
@@ -134,8 +134,13 @@
 
         <div id="assessment" class="collapse">
             <label for="assessmentDay">Assessment day</label>
+            <p>
+                <label for="pastAssessmentDays">
+                    <input type="checkbox" id="pastAssessmentDays" name="accept" value="yes">  Include past assessment days
+                </label>
+            </p>
+            <p>Current saved date: <?php echo $applicant->getAssessmentDay();?></p>
```
Comment (ashleycoles) [inline templates/applicantForm.phtml:142]: Please could you only display this if there is a date currently set? Also, wondering if it's possible only to display it if the date is a past one?

**Review by ashleycoles — CHANGES_REQUESTED** [review]
One last request!

**Review by ashleycoles — APPROVED** [review]

**Review by ashleycoles — APPROVED** [review]

**Pull request merged**